hooks/environment

```diff
@@ -1,2 +1,34 @@
-#!/bin/bash
-python "${BASH_SOURCE%/*}/../python/setup-hooks.py"
+#!/usr/bin/env bash
+
+declare -a hooks=(
+    # "environment" # not allowed
+    "pre-checkout"
+    "checkout"
+    "post-checkout"
+    "pre-command"
+    "command"
+    "post-command"
+    "pre-artifact"
+    "post-artifact"
+    "pre-exit"
+)
+declare -a extensions=(
+    ""
+    ".bat"
+    ".cmd"
+)
+hooks_dir="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+vars="$(mktemp)"
+env | sort | grep "METAHOOK" | uniq >"${vars}"
+for hook_name in "${hooks[@]}"; do
+  for ext in "${extensions[@]}"; do
+    hook_file="${hooks_dir}/${hook_name}${ext}"
+    rm -f "${hook_file}"
+    upperd="$(echo "${hook_name}${ext}" | tr "[:lower:]" "[:upper:]")"
+    bkv="BUILDKITE_PLUGIN_METAHOOK_${upperd//-/_}"
+    echo "bkv: ${bkv}"
+    if grep "${bkv}" <"${vars}"; then
+      echo "${!bkv}" >"${hook_file}"
```

Comment:
You might want to add a note here as indirect variable expansion is a useful but rarely used feature that only a handful of people will know without looking it up.